pkg/ingress/gateway.go

```diff
@@ -17,66 +16,10 @@ import (
 	"github.com/openservicemesh/osm/pkg/certificate"
 )
 
-// provisionIngressGatewayCert does the following:
-// 1. If an ingress gateway certificate spec is specified in the MeshConfig resource, issues a certificate
-//    for it and stores it in the referenced secret.
-// 2. Starts a goroutine to watch for changes to the MeshConfig resource and certificate rotation, and
-//    updates/removes the certificate and secret as necessary.
-func (c *client) provisionIngressGatewayCert(stop <-chan struct{}) error {
-	defaultCertSpec := c.cfg.GetMeshConfig().Spec.Certificate.IngressGateway
-	if defaultCertSpec != nil {
-		// Issue a certificate for the default certificate spec
-		if err := c.createAndStoreGatewayCert(*defaultCertSpec); err != nil {
```

Comment:
By creating the certificate before subscribing to updates, we were able to miss cert rotations